common/scala/src/main/scala/whisk/common/Logging.scala

```diff
@@ -21,79 +21,77 @@ import java.time.Clock
 
+trait Logging {
     /**
      * Prints a message on DEBUG level
      *
      * @param from Reference, where the method was called from.
      * @param message Message to write to the log
      */
-    def debug(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) =
-        if (level >= DebugLevel) emit(DebugLevel, id, from, message)
+    def debug(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = emit(DebugLevel, id, from, message)
 
     /**
      * Prints a message on INFO level
      *
      * @param from Reference, where the method was called from.
      * @param message Message to write to the log
      */
-    def info(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) =
-        if (level >= InfoLevel) emit(InfoLevel, id, from, message)
+    def info(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = emit(InfoLevel, id, from, message)
 
     /**
      * Prints a message on WARN level
      *
      * @param from Reference, where the method was called from.
      * @param message Message to write to the log
      */
-    def warn(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) =
-        if (level >= WarningLevel) emit(WarningLevel, id, from, message)
+    def warn(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = emit(WarningLevel, id, from, message)
 
     /**
      * Prints a message on ERROR level
      *
      * @param from Reference, where the method was called from.
      * @param message Message to write to the log
      */
-    def error(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) =
-        if (level >= ErrorLevel) emit(ErrorLevel, id, from, message)
+    def error(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = emit(ErrorLevel, id, from, message)
 
-    private var level = InfoLevel
-}
-
-/**
- * Facility that is used by Logger to write down the log-output to the standard output stream.
- */
-trait PrintStreamEmitter {
     /**
-     * The output stream, defaults to Console.out.
-     * This is mutable to allow unit tests to capture the stream.
-     */
-    var outputStream: PrintStream = Console.out
-
-    /**
-     * Prints a message to the output stream
+     * Prints a message to the output.
      *
      * @param loglevel The level to log on
      * @param id <code>TransactionId</code> to include in the log
      * @param from Reference, where the method was called from.
      * @param message Message to write to the log
-     * @param marker A LogMarkerToken. They are defined in <code>LoggingMarkers</code>.
      */
-    def emit(loglevel: LogLevel, id: TransactionId, from: AnyRef, message: String, marker: Option[LogMarker] = None) = {
+    def emit(loglevel: LogLevel, id: TransactionId, from: AnyRef, message: String)
```

Comment:
I moved the integration of the markers into `TransactionId`.
If it keeps here, all implementations of this trait need to handle the markers.